library/Icinga/Util/TimezoneDetect.php

```diff
@@ -51,30 +37,11 @@ public function __construct()
         }
 
         if (array_key_exists(self::$cookieName, $_COOKIE)) {
-            $matches = array();
-            if (preg_match('/\A(-?\d+)[\-,](\d+)\z/', $_COOKIE[self::$cookieName], $matches)) {
-                $offset = $matches[1];
-                $timezoneName = timezone_name_from_abbr('', (int) $offset, (int) $matches[2]);
-
-                self::$success = (bool) $timezoneName;
-                if (self::$success) {
-                    self::$offset = $offset;
-                    self::$timezoneName = $timezoneName;
-                }
-            }
+            self::$timezoneName = $_COOKIE[self::$cookieName];
```

Comment:
Previously only parts of the value were used and those were verified to only consist of digits.

Now the value is used as-is, which opens up a potential vulnerability. So please make sure we're dealing with a timezone name here. (e.g. set to verify)